src/pgtkterm.c

```diff
@@ -6625,16 +6621,17 @@ pgtk_begin_cr_clip (struct frame *f)
   cairo_t *cr = FRAME_CR_CONTEXT (f);
 
   PGTK_TRACE("pgtk_begin_cr_clip");
-  if (! FRAME_CR_SURFACE (f))
-    {
-      FRAME_CR_SURFACE(f) = gdk_window_create_similar_surface(gtk_widget_get_window (FRAME_GTK_WIDGET (f)),
-							      CAIRO_CONTENT_COLOR_ALPHA,
-							      FRAME_PIXEL_WIDTH (f),
-							      FRAME_PIXEL_HEIGHT (f));
-    }
-
   if (!cr)
     {
+      cairo_surface_t *surface =
+	gdk_window_create_similar_surface(gtk_widget_get_window (FRAME_GTK_WIDGET (f)),
+					  CAIRO_CONTENT_COLOR_ALPHA,
+					  FRAME_CR_SURFACE_DESIRED_WIDTH (f),
+					  FRAME_CR_SURFACE_DESIRED_HEIGHT (f));
+
+      cr = FRAME_CR_CONTEXT (f) = cairo_create (surface);
+      cairo_surface_destroy (surface);
+
       cr = cairo_create (FRAME_CR_SURFACE (f));
       FRAME_CR_CONTEXT (f) = cr;
```

Comment:
cairo_create is called twice, and cr_context is overwritten.

Reply:
Good Catch, I was clearly looking at/chaning that code for too long to see what I had done